### PR TITLE
fix(azure-iot-device-amqp): receiver links need autoaccept to be true…

### PR DESCRIPTION
**Pull request by anthonyvercolano**
… when messages come settled

### CODE REVIEW

File: device/transport/amqp/src/amqp_twin_client.ts [hunk 1]
```diff
@@ -161,7 +161,10 @@ export class AmqpTwinClient extends EventEmitter {
                   } else {
                     this._senderLink = senderTransportObject;
                     this._senderLink.on('error', this._errorHandler);
-                    this._client.attachReceiverLink( this._endpoint, this._generateTwinLinkProperties(linkCorrelationId), (receiverLinkError?: Error, receiverTransportObject?: any): void => {
+                    const linkOptions = this._generateTwinLinkProperties(linkCorrelationId);
+                    const autoAcceptPropertyName = 'autoaccept';
+                    linkOptions[autoAcceptPropertyName] = true;
```
Comment (YoDaMa) [inline device/transport/amqp/src/amqp_twin_client.ts:166]: Is there an absolute need to manipulate this object or can we adjust this when it's created.

File: device/transport/amqp/src/amqp_device_method_client.ts [hunk 2]
```diff
@@ -104,6 +104,8 @@ export class AmqpDeviceMethodClient extends EventEmitter {
                     this._fsm.transition('detaching', attachCallback, err);
                   } else {
                     this._senderLink = senderLink;
+                    const acceptProperty = 'autoaccept';
```
Comment (YoDaMa) [inline device/transport/amqp/src/amqp_device_method_client.ts:107]: Is this line necessary or can you just set this string as the key?

**Review by YoDaMa — APPROVED** [review]
See nitpicks on functional programming + unnecessary variable

File: device/transport/amqp/src/amqp_twin_client.ts (hunk 1 above)
Comment (anthonyvercolano) [reply device/transport/amqp/src/amqp_twin_client.ts:166]: Fixed.  Thanks for calling me out on that.

File: device/transport/amqp/src/amqp_device_method_client.ts (hunk 2 above)
Comment (anthonyvercolano) [reply device/transport/amqp/src/amqp_device_method_client.ts:107]: Fixed.

**Comment by scelle** [comment]
@anthonyvercolano 
sorry for abusing this channel,   I am using **node-red-contrib-azureiothub** (v0.5.3) but can't find a channel where I can report a problem or find solutions.  my node stops receiving c2d messages from time to time.  I read about a fix in the azure-iot-sdk (v1.17.0) and I was wondering if it might be related, if there's an update of the node-red-contrib-azureiothub planned, 

kind regards
Pascal Desmet

**Pull request merged**